e2e/atlas/cluster_generator_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build e2e || atlas
+
+package atlas_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"github.com/mongodb/mongocli/e2e"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type clusterGenerator struct {
```

Comment:
it may be a good idea to `godoc` this struct and its method so people understand how to use them

Reply:
also seeing as this is used for the cluster tests the name cluster generator adds some noise, some alternatives to consider, `TestCtx`, `AtlasTestCtx`, `AtlasE2ETest`, or something like it